projects/batfish-client/src/main/java/org/batfish/client/Client.java

```diff
@@ -1946,29 +2034,24 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
       throw new BatfishException("Failed to visit questions dir", e);
     }
     for (Path jsonQuestionFile : jsonQuestionFiles) {
-      int numBefore = bfq.size();
-      String name = loadQuestion(jsonQuestionFile, bfq);
-      int numAfter = bfq.size();
-      if (numBefore == numAfter) {
-        ae.getReplaced().add(name);
-      } else {
-        ae.getAdded().add(name);
+      try {
+        JSONObject questionJSON = loadQuestion(jsonQuestionFile, null, null);
+        String questionName =
+            questionJSON
+                .getJSONObject(BfConsts.PROP_INSTANCE)
+                .getString(BfConsts.PROP_INSTANCE_NAME);
+        if (localQuestions.containsKey(questionName.toLowerCase())) {
+          ae.getReplaced().add(questionName);
+        } else {
+          ae.getAdded().add(questionName);
+        }
+        localQuestions.put(questionName.toLowerCase(), questionJSON.toString());
+        numLoaded++;
+      } catch (JSONException e) {
+        throw new BatfishException("Could not find instanceName in question " + jsonQuestionFile);
```

Comment:
looks like a lot of code redundancy in this PR. Are there useful helpers to extract that would make the code much simpler and more understandable?

Reply:
Refactored and split the resusable parts in different more general functions, added corresponding tests for each function in client test